### DISCUSSION TITLE
Importing recipes and ingredient sections

### DISCUSSION THREAD

**Discussion by shimizurei** [mealie-recipes/mealie#2579] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

When you import a recipe with sections, it adds the section header as an ingredient. Then when you go to edit it into a section, you're left with an empty checkbox where the old text was. Deleting the list item results in the section's deletion...

![mealie](https://github.com/mealie-recipes/mealie/assets/3500094/98cef51e-cbd1-454f-b310-cdf1d5ded23a)

(Small note to the maintainers, "tagged" is misspelled as "taged" in the template)

### Please provide a concise description of the feature that would resolve your issue.

Can't we just create a section header? Or at least don't show empty list items?

### Please consider and list out some caveats or tradeoffs made in your design decision

All the ingredients might not be tied to the section...

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
Can you share the URL for the recipe in your example? Depending on where the issue lies it might be fixable, but I have a feeling it's a data issue

**Reply by shimizurei** [reply]
Link: https://www.seriouseats.com/korean-style-fire-chicken-buldak-with-cheese

**Comment by michael-genson** [comment]
Also:

> (Small note to the maintainers, "tagged" is misspelled as "taged" in the template)

Thanks! We'll get that fixed

**Comment by michael-genson** [comment]
> Link: https://www.seriouseats.com/korean-style-fire-chicken-buldak-with-cheese

Thanks! Yup, this is a source data issue, it lists the sections as ingredients. See the JSON here:
```json
"recipeIngredient": [
    "For the Sauce:",
    "1/2 cup (120ml) homemade brown chicken stock or store-bought, low-sodium chicken broth",
    "1/2 cup (120ml) soy sauce",
    "1/4 cup (60ml) Korean rice syrup (see note)",
    "1/4 cup (60ml; 60g) gochujang (Korean chili paste)",
    "1/4 cup (25g) fine ground gochugaru (Korean chili powder)",
    "2 tablespoons (30ml) toasted sesame oil",
    "2 tablespoons (30ml) vegetable oil",
    "8 large garlic cloves, minced",
    "One (1-inch) piece fresh ginger, peeled and finely grated (about 1 tablespoon; 12g)",
    "2 fresh red chili peppers, minced (see note)",
    "1 tablespoon (6g) freshly ground black pepper",
    "1 cup (240ml) Sprite",
    "For the Chicken:",
    "4 bone-in, skin-on chicken thighs (about 1 1/2 pounds; 675g)",
    "To Finish:",
    "6 ounces (170g) low-moisture mozzarella, shredded",
    "2 scallions, thinly sliced",
    "1 tablespoon (6g) toasted sesame seeds"
]
```

Unfortunately there's nothing really we can do about that, it has to be manually edited. I've noticed with serious eats that most of their recipes are like that.
